src/dataflow/src/render/threshold.rs

```diff
@@ -37,38 +57,19 @@ where
             all_columns.push(expr::MirScalarExpr::Column(column));
         }
         input = input.ensure_arrangements(Some(all_columns.clone()));
-        match input
+        let arrangement = input
             .arrangement(&all_columns)
-            .expect("Arrangement ensured to exist")
-        {
-            ArrangementFlavor::Local(oks, errs) => {
-                let oks = oks.reduce_abelian::<_, OrdValSpine<_, _, _, _>>(
-                    "Threshold",
-                    move |_k, s, t| {
-                        for (record, count) in s.iter() {
-                            if *count > 0 {
-                                t.push(((*record).clone(), *count));
-                            }
-                        }
-                    },
-                );
-                CollectionBundle::from_columns(0..arity, ArrangementFlavor::Local(oks, errs))
-            }
-            ArrangementFlavor::Trace(_, oks, errs) => {
-                let oks = oks.reduce_abelian::<_, OrdValSpine<_, _, _, _>>(
-                    "Threshold",
-                    move |_k, s, t| {
-                        for (record, count) in s.iter() {
-                            if *count > 0 {
-                                t.push(((*record).clone(), *count));
-                            }
-                        }
-                    },
-                );
-                use differential_dataflow::operators::arrange::ArrangeBySelf;
-                let errs = errs.as_collection(|k, _| k.clone()).arrange_by_self();
-                CollectionBundle::from_columns(0..arity, ArrangementFlavor::Local(oks, errs))
-            }
-        }
+            .expect("Arrangement ensured to exist");
+        let negatives = match &arrangement {
+            ArrangementFlavor::Local(oks, _) => threshold_arrangement(oks),
+            ArrangementFlavor::Trace(_, oks, _) => threshold_arrangement(oks),
+        };
+        let (oks, errs) = arrangement.as_collection();
+        let oks = negatives
+            .as_collection(|k, _| k.clone())
+            .negate()
+            .concat(&oks)
+            .consolidate();
+        CollectionBundle::from_collections(oks, errs)
```

Comment:
Same comment as with `distinct`: given that these are not massive, we'd probably like both implementations and an enum in the plan that allows the planner to toggle between the two as it sees fit (and then we could determine the best defaults).

Reply:
Same here: both variants are around now and can be selected through the plan.